### DISCUSSION TITLE
New Drone with custom specifications

### DISCUSSION THREAD

**Discussion by Gianluca-Coppola** [CodexLabsLLC/Colosseum#48] [Q&A]
Hi, im a student using AirSim for my thesis project, currently still trying to analyze and figure out the whole architecture of the plugin.
My goal for now is to make a drone with custom specifications, I examined a few classes related to the Drone specs, like MultiRotorParams and RotorParams, but I dont quite understand where these classes are used in the code, and how I could extend them to fit my custom criteria.
So far I managed to add AirSim to my unreal environment, and can move the drone using simple api calls with simple_flight as my Flight Controller.
I apologize if this question might be trivial, I dont have a lot of experienced with Unreal Engine nor coding in C++ 
Thanks

**Comment by tenxlenx** [comment]
hey i happen to write my thesis and a big part of it is Airsim and It took me a lot of time to figure out how stuff works, i think once you get how the project is structured it does make sense and I can give you some help by providing you some examples.
If your plan is for the custom drones is to add sensors and cameras of all types, it is quite easy and you can just use the binaries compiled so in this case you don't need UE as you can configure your drones in .json file. 
Those binaries what you download that is your "server" what you compile if you do the UE build - this is only necessary if you want to add a new sensor, new type of vehicle, or a custom environment. Your "client" application is basically the c++ or python project interfacing with the Airsim UE plugin via rpclib. (or in your case the shell script) The HelloCar and HelloDrone examples are quite helpful, try to get them running and it will be a good starting point.. If you could say a few more info what are you trying to do, i can try my best to help you

**Reply by Gianluca-Coppola** [reply]
Hi, thanks for getting back to me. 
My idea of a custom drone is being able to set a different weight, battery size, thrust force etc.. , so the End-user of my application might configure their ideal drone for the simulation.
I did move away from simple_flight as a flight controller, Im now trying to work with PX4 and QGroundControl. For now I just managed to connect them, the next step will still be try to implement said custom drone using PX4

**Comment by Eric-Carleton** [comment]
I am also a student using AirSim as a major part of my thesis, and I am also hoping for a way to configure the size/mass/shape characteristics of the quadrotor... and potentially even configure it to be a hexacoptor or octacopter if that is at all possible (maybe even a fixed wing?). 

I haven't had a chance to test this yet, but this previous article from AirSim has some information on changing mass/performance parameters: https://github.com/microsoft/AirSim/issues/3311

and this previous article from AirSim has a tutorial on how to configure a hexacopter when using the PX4: https://github.com/Microsoft/AirSim/wiki/hexacopter

I have yet to find a single post that explains the full process though. Not sure if anyone has done this before?
